src/main/java/build/buildfarm/proxy/http/HttpBlobStore.java

```diff
@@ -374,10 +374,10 @@ private void releaseDownloadChannel(Channel ch) {
   }
 
   private boolean isChannelPipelineEmpty(ChannelPipeline pipeline) {
-    return (pipeline.first() == null)
-        || (useTls
-            && "ssl-handler".equals(pipeline.firstContext().name())
-            && pipeline.first() == pipeline.last());
+    return (pipeline.first() != null)
+        && (!useTls
+            || !"ssl-handler".equals(pipeline.firstContext().name())
+            || pipeline.first() != pipeline.last());
```

Comment:
function name needs changed.  maybe `isChannelFull`..  something like that

Reply:
Just going to revert here.